next.config.js

```diff
@@ -26,6 +30,49 @@ module.exports = withCSS(withSourceMaps({
     APP_VERSION: packageJson.version,
     UNLY_SIMPLE_LOGGER_ENV: process.env.APP_STAGE, // Used by @unly/utils-simple-logger - Fix missing staging logs because it believes we're in production
   },
+  experimental: {
+    redirects() {
+      const redirects = [
+        {
+          // Redirect root link with trailing slash to non-trailing slash, avoids 404 - See https://github.com/zeit/next.js/discussions/10651#discussioncomment-8270
+          source: '/:locale/',
+          destination: '/:locale',
+          permanent: process.env.APP_STAGE !== 'development', // Do not use permanent redirect locally to avoid browser caching when working on it
+        },
+        {
+          // Redirect link with trailing slash to non-trailing slash (any depth), avoids 404 - See https://github.com/zeit/next.js/discussions/10651#discussioncomment-8270
+          source: '/:locale/:path*/',
+          destination: '/:locale/:path*',
+          permanent: process.env.APP_STAGE !== 'development', // Do not use permanent redirect locally to avoid browser caching when working on it
+        },
+      ];
+
+      if (process.env.APP_STAGE === 'development') {
+        console.info('Using experimental redirects:', redirects);
+      }
+
+      return redirects;
+    },
+    rewrites() {
+      const rewrites = [
+        {
+          // XXX Doesn't work locally (maybe because of rewrites), but works online
+          source: '/',
+          destination: '/api/autoRedirectToLocalisedPage',
+        },
+        {
+          source: `/:locale((?!${allowedLocales.join('|')})[^/]+)(.*)`,
```

Comment:
Hello! I have a similar solution, but it does not work fully.
Example: if you have `en` locale you can pass `enduro` and this rewrite will not work.
[Codesandbox example](https://codesandbox.io/s/sharp-lewin-6iuqg?file=/src/index.js:0-321)
Check also this path-to-regexp issue https://github.com/pillarjs/path-to-regexp/issues/223

Reply:
Hi @mxck, I'm not sure I get your point here. Why would you want `enduro` to redirect to `en`? I prefer the current behaviour, which matchs against the exact locale name.

Reply:
Sorry, I probably didn’t explain it well :)
`enduro` should be redirected to `en/enduro`, but right now it not catches by rewrites

Reply:
Indeed! I get it now.
Maybe it's just a regex pattern issue. I was inspired by codesandbox.io/s/nextjs-i18n-staticprops-new-ouyrb

Regexes really aren't my thing, so I'll let it slide until someone can propose a fix.

A potential workaround is to use fr-FR and en-US instead of fr and en.